luigi/parameter.py

```diff
@@ -311,13 +311,19 @@ class DateParameter(_DateParameterBase):
     """
 
     date_format = '%Y-%m-%d'
-    _timedelta = timedelta(days=1)
 
-    def parse(self, s):
-        """
-        Parses a date string formatted as ``YYYY-MM-DD``.
-        """
-        return datetime.datetime.strptime(s, self.date_format).date()
+    def next_in_enumeration(self, value):
+        return value + datetime.timedelta(days=self.interval)
+
+    def normalize(self, value):
+        if value is None:
+            return None
+
+        if isinstance(value, datetime.datetime):
+            value = value.date()
```

Comment:
Yep, I think so. It sort of worked out via duck-typing before (i.e. datetime has .year/.month/etc), but now there's a far smaller chance of introducing bugs by accidentally using a date as a datetime (and not noticing cause you always passed in a datetime)